src/connection-manager/index.ts

```diff
@@ -649,7 +649,15 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
 
     for (const connection of sortedConnections) {
       log('too many connections open - closing a connection to %p', connection.remotePeer)
-      toClose.push(connection)
+      // check allow list
+      const connectionInAllowList = this.allow.some((ma) => {
+        return ma.getPeerId() === connection.remotePeer.toString()
```

Comment:
In the test, `allow` is specified as `/ip4/83.13.55.32` - it has no peer id so this check doesn't look right.

I think instead we should be comparing address prefixes here, similar to the `acceptIncomingConnection` method in this class:

```suggestion
        return connection.remoteAddr.toString().startsWith(ma.toString())
```